drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -568,7 +568,62 @@ impl<'a> Handler {
                 )?;
                 Ok((RspBody::Ok, None))
             }
+            ReqBody::State(StateReq::LifecycleState) => {
+                let out = self.lifecycle_state();
+
+                Ok((RspBody::State(out.map(StateRsp::LifecycleState)), None))
+            }
+        }
+    }
+
+    fn lifecycle_state(&mut self) -> Result<LifecycleState, StateError> {
+        const CMPA_SIZE: usize = 512;
+        let mut buf = [0u8; CMPA_SIZE];
+
+        // If the SHA-256 Digest is zeros, the CMPA is unlocked.  The SHA-256
+        // digest is located in the last 4 words of the CMPA.
+        self.update
+            .read_rot_page(RotPage::Cmpa, &mut buf)
+            .map_err(StateError::ReadCmpa)?;
+        if buf[CMPA_SIZE - 32..].iter().all(|b| *b == 0) {
+            // We use Unprogrammed as a catchall if the CMPA is unlocked
+            return Ok(LifecycleState::Unprogrammed);
```

Comment:
No need to do this here but I think we may want to consider abstracting this out for future use